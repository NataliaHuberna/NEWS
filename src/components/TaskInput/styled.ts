import styled from "styled-components";
import {COLORS} from '../../constants/colors';

export const StDiv = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 500px;
  min-height: 40px;
  background-color: ${COLORS.darkgoldenrod};
  border-radius: 5px;
  padding: 10px;
  margin-top: 40px;
`;

export const StButton = styled.button`  
    width: 15%;
    background-color: ${COLORS.blue};
    color: ${COLORS.white};
    outline: none;
    border: none;
    border-radius: 5px;
    height: 25px;
    cursor: pointer;
`;

export const StInput = styled.input`
  width: 75%;
  height: 25px;
  border-radius: 5px;
  border-color: transparent;
  padding: 0 10px
`;