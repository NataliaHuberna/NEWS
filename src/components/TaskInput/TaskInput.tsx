import React,  { useState, useEffect } from 'react';
import {StDiv, StButton, StInput} from "./styled";

const TaskInput = ({addTodos}) => {
    const addTask = (task) => {
        const newTask = {value: task, id: Date.now(), checked: false};
        addTodos(newTask);
    };
    const inputRef = React.createRef();
    const[input, setInput] = React.useState('');

    useEffect(()=> inputRef.current.focus());

    const addTaskToList = () => {
        if (input.trim()) {
            addTask(input);
            setInput( '');
        }
    };

    const handleEnter = (event) => {if (event.key === 'Enter') addTaskToList();};
    const inputChange = (event) => setInput(event.target.value );

    return (
        <StDiv>
            <StInput
                ref={inputRef}
                onKeyPress={handleEnter}
                onChange={inputChange}
                value={input}
                placeholder="Add new task..."
            />
            <StButton disabled={!input.trim()} onClick={addTaskToList}>
                ADD
            </StButton>
        </StDiv>
    );
};

export default TaskInput;
