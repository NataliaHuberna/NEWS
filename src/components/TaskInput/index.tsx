import {connect} from 'react-redux';
import {addTodos} from "../../store/todos/actions";
import TaskInput from "./TaskInput";

const mapDispatchToProps = dispatch => ({
    addTodos: (newTask) => dispatch(addTodos(newTask))
});

export default connect(null, mapDispatchToProps)(TaskInput);