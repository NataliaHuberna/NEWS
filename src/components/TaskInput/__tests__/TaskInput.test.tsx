import React from "react";
import {mount, shallow} from 'enzyme';
import TaskInput from "../TaskInput";
import {act} from "react-dom/test-utils";

describe('TaskInput', () => {
    let props;
    const setStateMock = jest.fn();

    beforeEach(() => {
        props = {
            addTodos: jest.fn()
        }
        React.useState = jest.fn().mockReturnValue(['some title', setStateMock]);
        Date.now = jest.fn(() => 1487076708000);
    })

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should set in state', () => {
        const component = shallow(<TaskInput {...props}/>);
        expect(component).toMatchSnapshot();
    });

    it('should set to state', () => {
        const component = mount(<TaskInput {...props}/>);
        act(()=> {
            component.find('input').getElement().props.onChange({target:{ value : 'some text'}});
        });
        expect(setStateMock).toHaveBeenCalledWith('some text');
    });

    it ('should check prop add task', () => {
        const component = mount(<TaskInput {...props}/>);
        act(()=> {
            component.find('button').getElement().props.onClick();
        });
        expect(props.addTodos).toHaveBeenCalledWith({value: 'some title', id: Date.now(), checked: false});
        expect(setStateMock).toHaveBeenCalled();
    });

    it ('should check prop add task with enter', () => {
        const component = mount(<TaskInput {...props}/>);
        act(()=> {
            component.find('input').simulate('keypress', { key: 'Enter' });
        });
        expect(props.addTodos).toHaveBeenCalledWith({value: 'some title', id: Date.now(), checked: false});
        expect(setStateMock).toHaveBeenCalled();
    });

})