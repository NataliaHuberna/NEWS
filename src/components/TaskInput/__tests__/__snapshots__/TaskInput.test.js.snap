// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskInput should set in state 1`] = `ShallowWrapper {}`;
